### PR TITLE
Implement generic type for Stack

### PR DESCRIPTION
**Pull request by hiepnv90**

### CODE REVIEW

File: pkg/ds/stack/stack.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+// Len returns number of elements in the stack.
+func (s *Stack[T]) Len() int {
+	return len(s.elems)
+}
+
+// Peek views the top element of the stack.
+// nolint: ireturn
+func (s *Stack[T]) Peek() T {
```
Comment (ngocthanh1389) [inline pkg/ds/stack/stack.go:30]: Should check empty stack

File: pkg/ds/stack/stack.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+// Push pushes a new element into the stack.
+func (s *Stack[T]) Push(elem T) {
+	s.elems = append(s.elems, elem)
+}
+
+// Pop pops the top element from the stack.
+// Panic if the stack is empty.
+// nolint: ireturn
+func (s *Stack[T]) Pop() T {
```
Comment (ngocthanh1389) [inline pkg/ds/stack/stack.go:42]: Shouldn't panic. Should return error, user will decide panic or error handle

Comment (hiepnv90) [reply pkg/ds/stack/stack.go:42]: I thought about it, not sure if it better to return error for leave the check for the caller.

File: pkg/ds/stack/stack.go (hunk 1 above)
Comment (hiepnv90) [reply pkg/ds/stack/stack.go]: Fixed https://github.com/KyberNetwork/tradinglib/pull/3/commits/26b7d6688a515421dd3d532c19f130cc5a9ba03f

**Review by secmask — APPROVED** [review]

**Pull request merged**